Gems/Atom/RHI/Code/Include/Atom/RHI.Reflect/ShaderStages.h

```diff
@@ -32,6 +32,10 @@ namespace AZ::RHI
         //! a vertex. On certain platforms like Metal, this stage may occur after tessellation.
         Vertex = 0,
 
+        //! This virtual stage contains shader stages that expand an input assembly stream and manipulate
+        //! a vertex.
+        Geometry,
```

Comment:
I would have preferred to add mesh shader support as that is better solution going forward. The problem with geometry is that it doesnt have metal support and that can have implications if people build a pipeline with geometry shader and expect it to work on all platforms. I have checked and all 3 backends (Vk/Dx12/Metal) have support for mesh shaders. Lets add a comment here that this stage is not supported on Metal.

Reply:
I checked and to add mesh shaders we would need D3DX12_MESH_SHADER_PIPELINE_STATE_DESC that is not available in the d3dx12 header we have. It is a crucial piece to add cross platform mesh shader support

Reply:
yeah. You will need DirectX agility sdk. There is a RFC related to that here - https://github.com/o3de/sig-graphics-audio/issues/154. Once this support is added adding mesh shaders support should be possible.

Reply:
Ok, so let's merge this for now and change to mesh shaders later.